### ISSUE TITLE
cmd/go/internal/vcweb/vcstest: `TestScripts/bzr` failure with `Unable to determine your name`

### ISSUE THREAD

**Issue by lacombar** [golang/go#62620]
### What version of Go are you using (`go version`)?
```go version go1.18.2 linux/amd64``` building `v1.21.1`

### Does this issue reproduce with the latest release?
this is the latest release

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
% go env
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE=<redacted>
GOENV=<redacted>
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE=<redacted>
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH=<redacted>
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/home/al/opt/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR=<redacted>
GOVCS=""
GOVERSION="go1.18.2"
GCCGO="gccgo"
GOAMD64="v1"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD=<redacted>
GOWORK=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build4266833550=/tmp/go-build -gno-record-gcc-switches"
</pre></details>

### What did you do?
build v1.21.1 via src/all.bash

### What did you expect to see?
```
ok cmd/go/internal/vcweb/vcstest
```

### What did you see instead?
```
2023/09/13 13:13:05 serving /
2023/09/13 13:13:07 serving /
--- FAIL: TestScripts (0.04s)
    --- FAIL: TestScripts/bzr/hello.txt (2.22s)
        vcstest_test.go:155: 2023/09/13 13:13:07 hello.txt:
            > handle bzr
            > env BZR_EMAIL='Russ Cox <rsc@google.com>'
            > bzr init-repo .
            [stdout]
            Shared repository with trees (format: 2a)
            Location:
              shared repository: .
            > bzr init b
            [stdout]
            Created a repository tree (format: 2a)
            Using shared repository: /tmp/TestScripts3796191388/001/bzr/hello/
            > cd b
            > cp ../hello.go .
            > bzr add hello.go
            [stdout]
            adding hello.go
            > bzr commit --commit-time='2017-09-21 21:20:12 -0400' -m 'hello world'
            [stderr]
            brz: ERROR: Unable to determine your name.
            Please, set your name with the 'whoami' command.
            E.g. brz whoami "Your Name <name@example.com>"


        vcstest_test.go:161: hello.txt:11: bzr commit '--commit-time=2017-09-21 21:20:12 -0400' -m 'hello world': exit status 3
FAIL
FAIL    cmd/go/internal/vcweb/vcstest   2.516s
```

**Comment by bcmills** [comment]
@lacombar, we need more detail in order to reproduce this.

The `BZR_EMAIL` environment variable in the test is supposed to prevent exactly this failure mode:
https://github.com/breezy-team/breezy/blob/9d812602812df574beb63ee13005f56f600aecb8/breezy/config.py#L544-L560

It isn't clear to me why that isn't having the intended effect for you. What version of the `bzr` command are you using? What relevant configuration do you have on your machine and/or in your environment?

**Comment by lacombar** [comment]
This is stock `bzr` from Ubuntu 2020

```% lsb_release -a 
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.6 LTS
Release:        20.04
Codename:       focal

% bzr --version                                            
Breezy (brz) 3.0.2
  Python interpreter: /usr/bin/python3 3.8.10
  Python standard library: /usr/lib/python3.8
  Platform: Linux-5.15.5-76051505-generic-x86_64-with-glibc2.29
  breezy: /usr/lib/python3/dist-packages/breezy
  Breezy configuration: /home/al/.config/breezy
  Breezy log file: /home/al/.brz.log

Copyright 2005-2012 Canonical Ltd.
https://www.breezy-vcs.org/

brz comes with ABSOLUTELY NO WARRANTY.  brz is free software, and
you may use, modify and redistribute it under the terms of the GNU
General Public License version 2 or later.```

**Comment by lacombar** [comment]
d1c783e2373badfba18a56d74dc8a6b0abd64431 fixes the issue locally. Do you want me to open a PR ?

**Comment by bcmills** [comment]
FWIW, it looks like this was fixed in `brz` in their 3.1.0 release:
https://github.com/breezy-team/breezy/commit/30143860ced0ad9f42cbf18a119f2800117decff

So you could also consider upgrading to that. (But setting `EMAIL` seems fine too.) You're welcome to open a PR.
